modules/apps/frontend-js/frontend-js-bundle-config-extender/src/main/java/com/liferay/frontend/js/bundle/config/extender/internal/servlet/taglib/JSBundleConfigTopHeadDynamicInclude.java

```diff
@@ -12,127 +12,97 @@
  * details.
  */
 
-package com.liferay.frontend.js.bundle.config.extender.internal;
+package com.liferay.frontend.js.bundle.config.extender.internal.servlet.taglib;
 
+import com.liferay.frontend.js.bundle.config.extender.internal.JSBundleConfigTracker;
 import com.liferay.petra.string.StringBundler;
 import com.liferay.portal.kernel.log.Log;
 import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.servlet.taglib.BaseDynamicInclude;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
 import com.liferay.portal.kernel.util.ContentTypes;
 import com.liferay.portal.kernel.util.Portal;
 import com.liferay.portal.kernel.util.StringUtil;
-import com.liferay.portal.minifier.MinifierUtil;
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.PrintWriter;
-import java.io.StringWriter;
 
 import java.net.URL;
 
 import java.util.Collection;
-import java.util.Map;
 
-import javax.servlet.Servlet;
-import javax.servlet.ServletConfig;
 import javax.servlet.ServletContext;
-import javax.servlet.ServletException;
-import javax.servlet.ServletOutputStream;
-import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import org.osgi.service.component.ComponentContext;
-import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Modified;
 import org.osgi.service.component.annotations.Reference;
 
 /**
- * @author Carlos Sierra Andrés
  * @author Chema Balsas
  */
 @Component(
-	immediate = true,
-	property = {
-		"osgi.http.whiteboard.servlet.name=com.liferay.frontend.js.bundle.config.extender.internal.JSBundleConfigServlet",
-		"osgi.http.whiteboard.servlet.pattern=/js_bundle_config",
-		"service.ranking:Integer=" + (Integer.MAX_VALUE - 1000)
-	},
-	service = {JSBundleConfigServlet.class, Servlet.class}
+	immediate = true, property = "service.ranking:Integer=" + Integer.MIN_VALUE,
```

Comment:
Giving this the `MIN_VALUE` as `service.ranking` will put it at the very end of the `/html/common/themes/top_js.jspf#resources` dynamic include, which is where the servlet was placed before (right after that) :+1:

Reply:
Also, looking at the other two instances of the `/html/common/themes/top_js.jspf#resources` dynamic include, both are `immediate` so it looks like a pattern. I'm not sure if it is really needed, because I think it should be pulled when the dynamic include manager navigates the services, but in any case, the result would be the same, so I guess we don't need to change it :shrug: